arex-instrumentation-foundation/src/main/java/io/arex/foundation/services/DataCollectorService.java

```diff
@@ -127,7 +129,12 @@ void saveData(DataEntity entity) {
         if (entity == null || CaseManager.isInvalidCase(entity.getRecordId())) {
             return;
         }
-        AsyncHttpClientUtil.postAsyncWithZstdJson(saveApiUrl, entity.getPostData(), null)
+        Map<String, String> requestHeaders = Maps.newHashMapWithExpectedSize(1);
```

Comment:
1. Use `MapUtils.newHashMapWithExpectedSize` instead of
2. Use `String.valueOf(object)` to avoid the NPE